src/tron/TronEngine.ts

```diff
@@ -216,20 +216,16 @@ export class TronEngine extends CurrencyEngine<TronTools, SafeTronWalletInfo> {
   async checkTokenBalances(): Promise<void> {
     const address = base58ToHexAddress(this.walletLocalData.publicKey)
 
-    for (const currencyCode of this.enabledTokens) {
-      const metaToken = this.allTokens.find(
-        token => token.currencyCode === currencyCode
-      )
-      if (metaToken?.contractAddress == null) continue
-      const contractAddressHex = base58ToHexAddress(metaToken.contractAddress)
-      const body = {
-        contract_address: contractAddressHex,
-        function_selector: 'balanceOf(address)',
-        parameter: padHex(address, 32),
-        owner_address: address
-      }
-
+    const balanceFuncs = this.enabledTokenIds.map(async tokenId => {
```

Comment:
This is not an array of functions. It's an array of promises.

I was very confused about how the code was working, but then I realized is was just named wrong.